lib/brakeman/report/report_codeclimate.rb

```diff
@@ -0,0 +1,70 @@
+require "json"
+require "yaml"
+
+class Brakeman::Report::CodeClimate < Brakeman::Report::Base
+  DOCUMENTATION_PATH = File.expand_path("../../../../docs/warning_types", __FILE__)
+  REMEDIATION_POINTS_CONFIG_PATH = File.expand_path("../config/remediation.yml", __FILE__)
+  REMEDIATION_POINTS_DEFAULT = 300_000
+
+  def generate_report
+    all_warnings.map { |warning| issue_json(warning) }.join("\0")
+  end
+
+  private
+
+  def issue_json(warning)
+    check = warning.check.sub("Brakeman::Check", "")
+    path = warning.file.sub("#{tracker.app_path}/", "")
+    body = content_for(warning.warning_type, warning.link)
+
+    {
+      type: "Issue",
+      check_name: check,
```

Comment:
@presidentbeef We use the `check_name` value to suppress issues in our UI. It's surfaced to users in that they must configure the `check_name` values that they'd like to suppress in their .codeclimate.yml file.

Suppressed checks are still run, we just don't surface the resulting issues to a user. Our thought for using the name for a `warning_code` here instead of the class name of a check is that since a check can emit one of several warning codes, it's safer to have a user intentional ignore each `warning_code` rather than an entire check. Does that make sense?